code/datums/diseases/robotic_transformation.dm

```diff
@@ -48,15 +69,40 @@
 			if (prob(8))
 				to_chat(affected_mob, "<span class='warning'>You can feel... something...inside you.</span>")
 		if(5)
-			to_chat(affected_mob, "<span class='warning'>Your skin feels as if it's about to burst off...</span>")
-			affected_mob.adjustToxLoss(10)
-			affected_mob.updatehealth()
-			if(prob(40)) //So everyone can feel like robot Seth Brundle
-				if(src.gibbed != 0) return 0
-				gibs(affected_mob)
-				gibbed = 1
-				var/mob/living/carbon/human/H = affected_mob
-				if(istype(H) && !jobban_isbanned(affected_mob, "Cyborg"))
-					H.Robotize()
-				else
-					affected_mob.death(1)
+			final_step("Cyborg")
+
+/datum/disease/robotic_transformation/mommi/stage_act()
```

Comment:
Oh dear, nobody ever taught you what polymorphism is.

1. Delete this entire thing
2. Make final step be two different procs for the cyborg and mommi transformation